File: Services/src/CodeRunner/CodeRunner.Model/CodeJobResult.cs
using System.Collections.Generic;

namespace CSC.CodeRunner.Model
{
	/// <summary>
	/// The result of a code job.
	/// </summary>
	public abstract class CodeJobResult
	{
		/// <summary>
		/// The status of the code job.
		/// </summary>
		public CodeJobStatus Status { get; set; }

		/// <summary>
		/// Diagnostic output.
		/// </summary>
		public string DiagnosticOutput { get; set; }

		/// <summary>
		/// The compilation result of the class.
		/// </summary>
		public CompilationResult ClassCompilationResult { get; set; }

		/// <summary>
		/// The compilation result for the tests, if the class compiled successfully.
		/// </summary>
		public CompilationResult TestsCompilationResult { get; set; }

		/// <summary>
		/// The results of the tests, if the tests compiled successfully.
		/// </summary>
		public List<CodeTestResult> TestResults { get; set; }
	}
}

File: Runners/JavaCodeRunner/src/main/java/csc/coderunner/CodeJobResult.java
package csc.coderunner;

import java.util.List;

/**
 * The result of a job.
 */
public abstract class CodeJobResult
{
	private CompilationResult classCompilationResult;
	private CompilationResult testsCompilationResult;
	private List<CodeTestResult> testResults;
	
	/**
	 * Constructor.
	 * @param classCompileResult The compilation result of the class.
	 * @param testsCompileResult The compilation result of the tests (if the class compiled successfully).
	 * @param testResults The test results (if the tests compiled successfully).
	 */
	public CodeJobResult(
		CompilationResult classCompileResult, 
		CompilationResult testsCompileResult, 
		List<CodeTestResult> testResults)
	{
		this.classCompilationResult = classCompileResult;
		this.testsCompilationResult = testsCompileResult;
		this.testResults = testResults;
	}
	
	/**
	 * @return The compilation result of the class.
	 */
	public CompilationResult getClassCompilationResult()
	{
		return classCompilationResult;
	}

	/**
	 * @return The compilation result of the tests (if the class compiled successfully).
	 */
	public CompilationResult getTestsCompilationResult()
	{
		return testsCompilationResult;
	}
	
	/**
	 * @return The test results (if the tests compiled successfully).
	 */
	public List<CodeTestResult> getTestResults()
	{
		return testResults;
	}
}
